src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -3101,16 +3231,78 @@ inline void rpp_resize_nn_load_i8pln1(Rpp8s *srcRowPtrsForInterp, Rpp32s *loc, _
 
 inline void rpp_store12_u8pkd3_to_u8pln3(Rpp8u* dstPtrR, Rpp8u* dstPtrG, Rpp8u* dstPtrB, __m128i &p)
 {
-    _mm_storeu_si128((__m128i *)(dstPtrR), _mm_shuffle_epi8(p, xmm_char_maskR)); /* Shuffle and extract the R pixels*/
-    _mm_storeu_si128((__m128i *)(dstPtrG), _mm_shuffle_epi8(p, xmm_char_maskG)); /* Shuffle and extract the G pixels*/
-    _mm_storeu_si128((__m128i *)(dstPtrB), _mm_shuffle_epi8(p, xmm_char_maskB)); /* Shuffle and extract the B pixels*/
+    _mm_storeu_si32((__m128i *)(dstPtrR), _mm_shuffle_epi8(p, xmm_char_maskR)); /* Shuffle and extract the R pixels*/
+    _mm_storeu_si32((__m128i *)(dstPtrG), _mm_shuffle_epi8(p, xmm_char_maskG)); /* Shuffle and extract the G pixels*/
+    _mm_storeu_si32((__m128i *)(dstPtrB), _mm_shuffle_epi8(p, xmm_char_maskB)); /* Shuffle and extract the B pixels*/
+}
+
+inline void rpp_store24_u8pkd3_to_u8pln3_avx(Rpp8u* dstPtrR, Rpp8u* dstPtrG, Rpp8u* dstPtrB, __m256i &p)
+{
+    __m128i p128[2];
+    const __m128i maskR1 = _mm_setr_epi8(0, 3, 6, 9, 12, 15, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80);
+    const __m128i maskG1 = _mm_setr_epi8(1, 4, 7, 10, 13, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80);
+    const __m128i maskB1 = _mm_setr_epi8(2, 5, 8, 11, 14, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80);
+    const __m128i maskR2 = _mm_setr_epi8(2, 5, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80);
+    const __m128i maskG2 = _mm_setr_epi8(0, 3, 6, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80);
+    const __m128i maskB2 = _mm_setr_epi8(1, 4, 7, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80);
+    p128[0] = _mm256_extractf128_si256(p, 0);
+    p128[1] = _mm256_extractf128_si256(p, 1);
+    _mm_storeu_si128((__m128i *)(dstPtrR), _mm_shuffle_epi8(p128[0], maskR1));
+    _mm_storeu_si128((__m128i *)(dstPtrG), _mm_shuffle_epi8(p128[0], maskG1));
+    _mm_storeu_si128((__m128i *)(dstPtrB), _mm_shuffle_epi8(p128[0], maskB1));
+
+    _mm_storeu_si128((__m128i *)(dstPtrR + 6), _mm_shuffle_epi8(p128[1], maskR2));
+    _mm_storeu_si128((__m128i *)(dstPtrG + 5), _mm_shuffle_epi8(p128[1], maskG2));
+    _mm_storeu_si128((__m128i *)(dstPtrB + 5), _mm_shuffle_epi8(p128[1], maskB2));
+}
+
+inline void rpp_store24_i8pkd3_to_i8pln3_avx(Rpp8s* dstPtrR, Rpp8s* dstPtrG, Rpp8s* dstPtrB, __m256i &p)
+{
+    __m128i p128[2];
+    const __m128i maskR1 = _mm_setr_epi8(0, 3, 6, 9, 12, 15, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80);
+    const __m128i maskG1 = _mm_setr_epi8(1, 4, 7, 10, 13, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80);
+    const __m128i maskB1 = _mm_setr_epi8(2, 5, 8, 11, 14, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80, 0x80);
```

Comment:
Same comment here for reader to understand masks - say as to why maskR1 has  6 Rs and maskR2 has 2 Rs etc

Reply:
Done